### PR TITLE
add vcs to the new created project

### PR DESCRIPTION
**Pull request by yuchunzhou**

### CODE REVIEW

File: nimble.nimble [hunk 1]
```diff
@@ -11,7 +11,7 @@ installExt = @["nim"]
 
 # Dependencies
 
-requires "nim >= 0.13.0"
+requires "nim >= 0.13.0", "gitapi"
```
Comment (SolitudeSF) [inline nimble.nimble]: nimble shouldn't depend on external packages.

File: src/nimble.nim [hunk 2]
```diff
@@ -858,6 +860,19 @@ js   - Compile using JavaScript backend.""",
     pkgRoot
   )
 
+  # Add git vcs to the new project
+  let repo = createGitRepo(path=pkgRoot)
+  discard repo.gitCommand("init")
+
+  var fd = open(joinPath(pkgRoot, ".gitignore"), fmWrite)
```
Comment (SolitudeSF) [inline src/nimble.nim]: library packages dont produce executables except for tests. and nothing ever produces `*.out` unless you explicitly do that.

File: src/nimble.nim [hunk 3]
```diff
@@ -858,6 +860,19 @@ js   - Compile using JavaScript backend.""",
...
+  fd.write(pkgName & "\n" & "*.out\n")
+  fd.close()
+
+  # Add a default xxx.nim.cfg file
+  fd = open(joinPath(pkgRoot, pkgSrcDir, pkgName & ".nim.cfg"), fmWrite)
+  fd.write("--d:release\n")
```
Comment (SolitudeSF) [inline src/nimble.nim]: you cant apply `release/danger` in `cfg` and it absolutely shouldn't be the default for no reason.

File: nimble.nimble (hunk 1 above)
Comment (yuchunzhou) [reply nimble.nimble]: ok, if the git init function is accepted, i can refactor the code later.

File: src/nimble.nim (hunk 3 above)
Comment (yuchunzhou) [reply src/nimble.nim]: @SolitudeSF why not specify a release version in the .nim.cfg file, i found that there are many nim users do not know how to input the compiler flags when compile a project, i thing it's a good beginning to create a default .nim.cfg when creating a new project.

Comment (SolitudeSF) [reply src/nimble.nim]: if they dont know about compiler flags, they for sure dont know about configuration files. and even if it was a good idea, it doesnt even work. you cant apply release flag from user configuration.

**Comment by dom96** [comment]
This is a cool idea, although I'm not sure we should be implicitly initializing a git repo when creating a new Nimble project. Some might not use git but mercurial instead.

How about you put these behind a ``--git`` flag?

**Comment by yuchunzhou** [comment]
@dom96 ok, i will try to update it.

File: src/nimble.nim (hunk 3 above)
Comment (yuchunzhou) [reply src/nimble.nim]: ok, i revert it.

Comment (Araq) [reply src/nimble.nim]: @SolitudeSF It's high time we fix that.

**Comment by yuchunzhou** [comment]
hi @dom96, i have added a `--vcs:git` option to init flag, the pull request supports git and hg both now, please review it.

File: src/nimble.nim (hunk 3 above)
Comment (yuchunzhou) [reply src/nimble.nim]: @SolitudeSF i think, if nim needs some compile flags, we should write them into a .cfg file, maybe it's not a good choice to write the compiler flags in the comand line. If all of us agree with the .cfg feature, i can add them before finish initializing a new project.

File: src/nimblepkg/options.nim [hunk 4]
```diff
@@ -80,6 +81,8 @@ Commands:
   init         [pkgname]          Initializes a new Nimble project in the
                                   current directory or if a name is provided a
                                   new directory of the same name.
+               [--vcs:git]        Add git or hg vcs to the new nimble project,
```
Comment (dom96) [inline src/nimblepkg/options.nim]: I would prefer explicit `--git` and `--hg` flags. Also ``git`` shouldn't be default, neither should be default.

File: src/nimble.nim [hunk 5]
```diff
@@ -858,6 +860,22 @@ js   - Compile using JavaScript backend.""",
     pkgRoot
   )
 
+  # Add vcs to the new nimble project
+  let vcsBin = options.action.vcsOption
+  if findExe(vcsBin, true).len != 0:
```
Comment (dom96) [inline src/nimble.nim]: This should be checked at the beginning of the function, if it doesn't exist then you should throw a NimbleError.

The actual logic you have here should still happen at the end though.

**Comment by yuchunzhou** [comment]
@dom96  ok, i updated the request's code and added `--git` and `--hg` flags for `nimble init` command.

File: src/nimble.nim (hunk 5 above)
Comment (yuchunzhou) [reply src/nimble.nim]: ok, i  put the check code to the beginning of the init function and the actual logic is still at the end of the init function

File: src/nimblepkg/options.nim (hunk 4 above)
Comment (yuchunzhou) [reply src/nimblepkg/options.nim]: yeah, i agree with you, i add both `--git` and `--hg` flags and if user inputs more than one flags, the later will overwrite the before ones.

**Comment by yuchunzhou** [comment]
@dom96 I find Nim is friendly to Windows platform, how about add a `--svn` flag to `nimble init` command?

**Pull request merged**